src/sage/graphs/generic_graph.py

```diff
@@ -21014,6 +21016,95 @@ def layout_graphviz(self, dim=2, prog='dot', **options):
         positions = dot2tex.dot2tex(self.graphviz_string(**options), format='positions', prog=prog)
 
         return {key_to_vertex[key]: pos for key, pos in positions.items()}
+    
+    def layout_tutte(self, external_face, external_face_pos=None, **options):
+        """
+
+        Compute graph layout based on a Tutte embedding.
+
+        The graph must be 3-connected and planar.
+
+        INPUT:
+
+        - ``external_face`` -- list; the external face to be made a polygon
```

Comment:
In order to calculate the embedding, an external face must be specified. Any subgraph of the graph that is a cycle will work as an external face. It would be possible to find an arbitrary external face automatically if none is specified, but it's unclear to me how one would be chosen from the available options. Also, that functionality is outside the scope of my original intention.

Reply:
Consider tha following:
```py
H = Graph(self) # take a (undirected) copy H of the graph
u, v = next(H.edge_iterator(labels=False)  # take any edge (u, v) of H
H.delete_edge(u, v)  # remove edge (u, v) from H
C = H.shortest_path(v, u) # Compute a shortest path from v to u in H minus (u, v)
```
You get a short cycle `C`, and this cycle is a face.

This can of course be done in a follow up PR, but this seems quite simple.